shell/bash

```diff
@@ -22,8 +22,17 @@
 #   ASH_LOG_BIN - the name of the binary that stores history in the db.
 #
 
-# Prevent errors from sourcing this file mor than once.
-[[ -n "${ASH_SESSION_ID}" ]] && return
+# Prevent errors from sourcing this file more than once.
+#
+# Under some circumstances ASH_SESSION_ID can be set but the functions not
+# defined properly, so we explicitly check for the __ash_* functions before
+# returning here.
+if [[ -n "${ASH_SESSION_ID}" ]]; then
```

Comment:
Thanks for the PR.  How does this look?
`[[ -n "${ASH_SESSION_ID}" ]] && declare -F __ash_log | grep -q . && return`